torch/_export/utils.py

```diff
@@ -34,7 +34,7 @@ def register_dataclass_as_pytree_node(
         cls
     ), f"Only dataclasses can be registered with this function: {cls}"
 
-    serialized_type = f"{cls.__module__}.{cls.__name__}"
+    serialized_type = f"{cls.__module__}.{cls.__qualname__}"
```

Comment:
I'm confused, is there overlap between `__qualname__` and `__module__`?

Reply:
> is there overlap between `__qualname__` and `__module__`?

No. The `__qualname__` refers to the fully qualified name. The `__qualname__` is different than `__name__` only in nested cases. Otherwise, they are the same.

[PEP 3155 – Qualified name for classes and functions](https://peps.python.org/pep-3155/#example-with-nested-classes)